lib/gprocess.c

```diff
@@ -683,7 +683,13 @@ g_process_change_limits(void)
   struct rlimit limit;
 
   getrlimit(RLIMIT_NOFILE, &limit);
+
+#if defined(__APPLE__) && defined(__MACH__)
+  limit.rlim_cur = MIN(OPEN_MAX, limit.rlim_max);
```

Comment:
Optional: a warning/info msg might be nice to have here, if the `OPEN_MAX` is less then `limit.rlim_max`.

Reply:
`rlim_max` is infinity by default, so the warning would be produced in most cases.